slicebase/bin/slice-canary

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+
+YUMCONF="yum.conf"
+CANARYCONF="yum.canary.conf"
+YUMREPOD="yum.slice.d"
+CANARYREPOD="yum.canary.d"
+
+if [ -s /etc/slicename ]; then
+    SLICENAME=$(cat /etc/slicename)
+else
+    echo "Error: /etc/slicename is either empty or doesn't exist."
+    exit 1
+fi
+
+if [[ $EUID -ne 0 ]] ; then
+    echo "Error: please run slice-canary as root."
+    exit 1
+fi
+
+if ! service slicectrl stop &> /dev/null; then
+    echo "Error: slicectrl failed to stop all slice services."
+    exit 1
+fi
+
+# Function to clean up temporary yum configs
+function cleanup {
+    rm -rf /tmp/$CANARYCONF /tmp/$CANARYREPOD
+}
+
+# Copy the regular production yum configurations to new "canary" versions that
+# we can use to upgrade the slice without actually modifying the production
+# versions.
+cp /etc/$YUMCONF /tmp/$CANARYCONF
+cp -a /etc/$YUMREPOD /tmp/$CANARYREPOD
+
+# Modify the "canary" yum configurations to point to the M-Lab staging yum
+# repository.
+sed -i "s/\/etc\/${YUMREPOD//\./\\.}/\/tmp\/${CANARYREPOD//\./\\.}/" \
```

Comment:
Sed supports alternate pattern delimiter characters. I think it may be whatever the first character is after "s". So for example, using `|` as the pattern delimiter, this should work:

    sed -i "s|/etc/yum.slice.d|/tmp/yum.canary.d|g" /tmp/yum.canary.conf

Also, I had to unpack what was going on with the variable expansions: `${YUMREPOD//\./\\.}`

If the shell escaping is necessary, I suggest creating variables prior to the sed command for improved readability. Is the escaping necessary?